File: src/server/pipelines/producer.go
package main

import (
    "bufio"
    "context"
    "fmt"
    "os"
    "time"
    "github.com/segmentio/kafka-go"
)

const (
    kafkaBroker = "127.0.0.1:9092"
    topic       = "data_pipeline"
)

func main() {
    writer := kafka.NewWriter(kafka.WriterConfig{
        Brokers:  []string{kafkaBroker},
        Topic:    topic,
        Balancer: &kafka.LeastBytes{},
    })

    file, err := os.Open("data/input.txt")
    if err != nil {
        panic("could not open input file " + err.Error())
    }
    defer file.Close()

    scanner := bufio.NewScanner(file)
    for scanner.Scan() {
        chunk := scanner.Text()
        err := writer.WriteMessages(context.Background(),
            kafka.Message{
                Key:   []byte(fmt.Sprintf("Key-%d", time.Now().UnixNano())),
                Value: []byte(chunk),
            },
        )
        if err != nil {
            panic("could not write message " + err.Error())
        }
        fmt.Println("Sent:", chunk)
    }

    if err := scanner.Err(); err != nil {
        panic("error reading input file " + err.Error())
    }

    writer.Close()
}

File: src/server/pipelines/producer.rs
use kafka::producer::{Producer, Record, RequiredAcks};
use std::fs::File;
use std::io::{BufRead, BufReader};
use std::time::Duration;

const KAFKA_BROKER: &str = "127.0.0.1:9092";
const TOPIC: &str = "data_pipeline";

fn main() {
    let mut producer = Producer::from_hosts(vec![KAFKA_BROKER.to_owned()])
        .with_ack_timeout(Duration::from_secs(1))
        .with_required_acks(RequiredAcks::One)
        .create()
        .unwrap();

    let file = File::open("data/input.txt").unwrap();
    let reader = BufReader::new(file);

    for line in reader.lines() {
        let chunk = line.unwrap();
        producer.send(&Record::from_value(TOPIC, chunk)).unwrap();
        println!("Sent: {}", chunk);
    }
}
